File: 2048-build-array-from-permutation/2048-build-array-from-permutation.cpp
class Solution {
public:
    vector<int> buildArray(vector<int>& nums) {
    int n=nums.size();
    vector<int> ans(n);
    for(int i: nums){
        int temp=nums[i];
        ans[i]=nums[temp];
    }
    return ans;
    }
};

File: 2048-build-array-from-permutation/2048-build-array-from-permutation.java
class Solution {
    public int[] buildArray(int[] nums) {
    int n=nums.length;
    int[] ans = new int[n];
    for(int i=0;i<n;i++){
        int temp= nums[i];
        ans[i]=nums[temp];
    } 
    return ans;  
    }
}